Rules/CommonScripts/Holiday/Birthday/Birthday.as

```diff
@@ -1 +1,48 @@
 // Birthday.as
+
+#include "BirthdayCommon.as";
+#include "HolidayCommon.as";
+
+void onInit(CRules@ this)
+{
+	if (!this.exists(holiday_head_prop))
+		this.set_u8(holiday_head_prop, 0);
+
+	if (!this.exists(holiday_head_texture_prop))
+		this.set_string(holiday_head_texture_prop, "BirthdayHeads.png");
+
+	// Also add balloons to map-initiated blobs
+	CBlob@[] blobs;
+	for (u8 i = 0; i < holiday_blobs.length(); i++)
+	{
+		getBlobsByName(holiday_blobs[i], @blobs);
+	}
+
+	for (u8 i = 0; i < blobs.length(); i++)
+	{
+		CBlob@ b = blobs[i];
+		AddBalloons(b.getSprite(), b.getName());
+	}
```

Comment:
Wont this need to be inside onRestart, since onInit is only called once on script is created (e.g. on first map load or after addScript). So unless the script get's removed and re-added every new round, this should be moved.